integration-cli/docker_cli_daemon_test.go

```diff
@@ -937,3 +955,99 @@ func TestDaemonRestartKillWait(t *testing.T) {
 
 	logDone("wait - wait on a stopped container doesn't timeout")
 }
+
+// TestHttpsInfo connects via two-way authenticated HTTPS to the info endpoint
+func TestHttpsInfo(t *testing.T) {
+	d := NewDaemon(t)
+	d.Start("--tlsverify", "--tlscacert", "fixtures/https/ca.pem", "--tlscert", "fixtures/https/server-cert.pem",
+		"--tlskey", "fixtures/https/server-key.pem", "-H", testDaemonHttpsAddr)
+	defer d.Stop()
+
+	cli := client.NewDockerCli(nil, ioutil.Discard, ioutil.Discard, "", testDaemonProto,
+		testDaemonHttpsAddr, getTlsConfig("client-cert.pem", "client-key.pem", t))
+
+	if msg := setTimeout("Reading command output time out", 10*time.Second, func() string {
+		if err := cli.CmdInfo(); err != nil {
+			return err.Error()
+		}
+		return ""
+	}); msg != "" {
+		t.Fatal(msg)
+	}
+
+	logDone("daemon - with tld enabled api server, good certificates")
+}
+
+// TestHttpsInfoRogueCert connects via two-way authenticated HTTPS to the info endpoint
+// by using a rogue client certificate and checks that it fails with the expected error.
+func TestHttpsInfoRogueCert(t *testing.T) {
+	d := NewDaemon(t)
+	d.Start("--tlsverify", "--tlscacert", "fixtures/https/ca.pem", "--tlscert", "fixtures/https/server-cert.pem",
+		"--tlskey", "fixtures/https/server-key.pem", "-H", testDaemonHttpsAddr)
+	defer d.Stop()
+
+	cli := client.NewDockerCli(nil, ioutil.Discard, ioutil.Discard, "", testDaemonProto,
```

Comment:
Stead of calling `client.NewDockerCLI`, this should use the `d.Cmd`, which allows you to construct a real CLI command connecting to the daemon you created.